src/components/i18n/i18n.tsx

```diff
@@ -64,9 +65,11 @@ function lookupToken<
     return children as RESOLVED;
   }
 
-  const Component: FunctionComponent<any> = () => {
-    return <Fragment>{children}</Fragment>;
-  };
+  const Component: FunctionComponent<any> = i18nComponentRenderFunc
+    ? i18nComponentRenderFunc(children)
+    : () => {
+        return <Fragment>{children}</Fragment>;
+      };
```

Comment:
optional implicit return style nit
```suggestion
    : () => <Fragment>{children}</Fragment>;
```